Metadata stability tests for modulemd metadata

dralley opened the pull request:
closes #2242
https://github.com/pulp/pulp_rpm/issues/2242

dralley commented:
Failure is a concurrency issue related to having two fixtures sharing the same modules but one with static context and one not, I think.  The tests work when run in isolation.  Will fix before merging

ggainey reviewed (approved):
This all looks reasonable to me - approved (w the caveat of "we need to fix the failing test before merging")

dralley commented:
@ggainey See new commit, I've merged a couple of tests together, and moved a few around, and simplified a bit.

Will rebase & delete commented code before merge

ggainey commented on pulp_rpm/tests/functional/constants.py:
```diff
@@ -1542,9 +1543,347 @@
-    "pulp_href",
-    "static_context",
-    "stream",
+    "pulp_created" "stream",
```
```suggestion
    "pulp_created",
     "stream",
```

dralley replied:
Nice

ggainey commented on pulp_rpm/tests/functional/constants.py:
```diff
@@ -1542,9 +1543,347 @@
+    "pulp_created" "stream",
     "version",
 ]
+
```
It would be great if we could have things like the following modulemds be static asset-files that the tests load, rather than having them embedded in the test-code. We have (iirc) similar embedding for advisory-tests. 

That's not something to do under this PR, just seeing them here brought to mind.

dralley replied:
Well for the publish test we do compare the fixture file directly against the Pulp file, but in this case we're looking at how the API output is structured, and the file can't help us with that.  Or do you mean we have JSON for the advisory tests laying around somewhere?

ggainey replied:
> Or do you mean we have JSON for the advisory tests laying around somewhere?

This - get the data out of the code. Among other things, makes it easier to produce/manipulate said data using JSON tools, than when it's embedded inside python.

Anyway - not a thing for this PR, just something that's been itching at me that these changes brought back to mind.

ggainey reviewed (changes requested):
Found one problem, otherwise looks great!
